### ISSUE TITLE
Bug: (ESLint ignores indentation rules in TypeScript interfaces)

### ISSUE THREAD

**Issue by AlexanderKosmehl** [eslint/eslint#16101]
### Environment

Node version: 16.5.1
npm version: 8.3.0
Local ESLint version: v8.19.0
Global ESLint version: v8.19.0
Operating System: Windows 11 (22000.739)

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```
{
    "env": {
        "browser": true,
        "es2021": true
    },
    "extends": [
        "standard"
    ],
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaVersion": "latest",
        "sourceType": "module"
    },
    "plugins": [
        "@typescript-eslint"
    ],
    "rules": {
    }
}
```
</details>

```ts
  export const hello = () => {
    console.log('Hello')
  }

  export interface User {
    name: string
    id: number
  }

  class User2 implements User {
    name: string
    id: number

    constructor() {
      this.name = 'Test'
      this.id = 15
    }
  }
```

### What did you expect to happen?

ESLint should detect indentation rule violation in all lines

### What actually happened?

ESLint detected intentation violations everywhere except in the interface code

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

I tested both standard and airbnb and the problem happened in both rulesets.

Screenshot of the issue:
https://imgur.com/a/FQccn4a

**Comment by mdjermanovic** [comment]
Hi @AlexanderKosmehl, thanks for the issue!

The core `indent` rule does not support TypeScript-specific syntax. You can use the [`@typescript-eslint/indent`](https://typescript-eslint.io/rules/indent) rule instead.
